Add functinonality to delete all comments.

eriyezab opened the pull request.

eriyezab committed:
Add functinonality to delete all comments.

shrutip90 commented on portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:
```diff
@@ -0,0 +1,49 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new comments. */
```
Update comment.

shrutip90 commented on portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:
```diff
@@ -0,0 +1,49 @@
+    PreparedQuery results = DATASTORE.prepare(QUERY);
+
+    for(Entity entity : results.asIterable()) {
+      //TODO: Remove later
+      System.out.println("Deleting comment made by: " + entity.getProperty("name"));
```
Looks like you want to remove this.

shrutip90 commented on portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:
```diff
@@ -0,0 +1,49 @@
+      DATASTORE.delete(entity.getKey());
+    }
+
+    // TODO: Remove later
+    System.out.println("Deleted all comments.");
```
ditto

shrutip90 commented on portfolio/src/main/webapp/comments.js:
```diff
@@ -1,3 +1,5 @@
+const COMMENTSLIST = document.getElementById('comments-list');
```
You should use COMMENTS_LIST since its two words. You could also call it COMMENTS_LIST_DOC_ELEMENT, to be more clear.

shrutip90 commented on portfolio/src/main/webapp/comments.js:
```diff
@@ -17,15 +19,31 @@ function getComments() {
+  if(deleteComments) {
+    fetch("/delete-comments", {method: "POST"});
+    removeCommentsFromPage()
+    //TODO: Remove later
+    console.log("Comments deleted.");
```
ditto. Although this one seems ok to keep.

eriyezab committed:
Remove comments marked with TODOs in DataServlet.java

eriyezab committed:
Update constant name and remove comment

updated the constant that holds the comments list's name to better reflect what it contains

shrutip90 commented on portfolio/src/main/webapp/comments.js:
```diff
@@ -17,15 +19,30 @@ function getComments() {
+  deleteComments = confirm("Are you sure you want to delete all comments? This action is irreversible!");
+  if(deleteComments) {
+    fetch("/delete-comments", {method: "POST"});
+    removeCommentsFromPage()
```
Doesn't this need a semi-colon at the end?

eriyezab committed:
Add missing semicolon to comments.js
